### ISSUE TITLE
Very old gradle plugin dependency 2.3.1 which is missing from maven

### ISSUE THREAD

**Issue by jeanhipages** [uxcam/react-native-ux-cam#93]
As of today (23 Jun 2023 9am Sydney time), we aren't able to build android in CI due to missing dependency for uxcam
`[01:48:46]: ▸ FAILURE: Build failed with an exception.
[01:48:46]: ▸ * What went wrong:
[01:48:46]: ▸ A problem occurred configuring project ':react-native-ux-cam'.
[01:48:46]: ▸ > Could not determine the dependencies of null.
[01:48:46]: ▸ > Could not resolve all task dependencies for configuration ':react-native-ux-cam:classpath'.
[01:48:46]: ▸ > Could not find com.android.tools.build:gradle:2.3.1.
[01:48:46]: ▸ Searched in the following locations:
[01:48:46]: ▸ - https://jcenter.bintray.com/com/android/tools/build/gradle/2.3.1/gradle-2.3.1.pom`

Could this dependency be updated?
https://github.com/uxcam/react-native-ux-cam/blob/main/uxcam-react-wrapper/android/build.gradle

**Comment by garkuwa** [comment]
same here, please help out

**Comment by KolissnikBogdan** [comment]
same

**Comment by gabrieluxcam** [comment]
Hello @jeanhipages @garkuwa @KolissnikBogdan is this happening on which specific SDK version? I've just tried to replicate it on 5.4.7 unsuccessfully, could you please let me know your SDK versions? Thanks!

**Comment by KolissnikBogdan** [comment]
@gabrieluxcam  
<img width="356" alt="Снимок экрана 2023-06-23 в 10 32 53" src="https://github.com/uxcam/react-native-ux-cam/assets/43568935/90d1a3a4-9f2a-4baa-a8ae-a5cb0f949539">

**Comment by KolissnikBogdan** [comment]
Maybe jcenter is currently unavailable...

**Comment by gabrieluxcam** [comment]
Thank you @KolissnikBogdan I'm checking this with our team and get back to you as soon as possible. Have you tested by cleaning the project? If this was working previously it may have to do with a connection issue or jcenter, as I have been able to build the app on my test project with multiple sdk versions. I will post more information as soon as I have it.

**Comment by KolissnikBogdan** [comment]
Quick solution:
replace com.android.tools.build:gradle:2.3.1 -> com.android.tools.build:gradle:2.3.0

**Comment by KolissnikBogdan** [comment]
https://repo.maven.apache.org/maven2/com/android/tools/build/gradle/
https://repo1.maven.org/maven2/com/android/tools/build/gradle/

<img width="778" alt="Снимок экрана 2023-06-23 в 10 38 12" src="https://github.com/uxcam/react-native-ux-cam/assets/43568935/c059866d-d75e-4e37-b216-a7f7faa2f9f0">

**Comment by KolissnikBogdan** [comment]
> 

I just ran the command: ./gradlew clean
can't say for sure if this issue is directly related to jscenter, but when I run the build I get similar errors from other SDKs

**Comment by KolissnikBogdan** [comment]
> Thank you @KolissnikBogdan I'm checking this with our team and get back to you as soon as possible. Have you tested by cleaning the project? If this was working previously it may have to do with a connection issue or jcenter, as I have been able to build the app on my test project with multiple sdk versions. I will post more information as soon as I have it.

I just ran the command: ./gradlew clean
can't say for sure if this issue is directly related to jscenter, but when I run the build I get similar errors from other SDKs

**Comment by gabrieluxcam** [comment]
Thanks for these details @KolissnikBogdan  if I get any confirmation or additional details from the team I will make sure to post them.

**Comment by ziga-hvalec** [comment]
any update on this? Or at least a workaround. Currently we can't build app on CI

**Comment by gabrieluxcam** [comment]
Hey @ziga-hvalec  have you tested @KolissnikBogdan 's current workaround? :

> Quick solution: replace com.android.tools.build:gradle:2.3.1 -> com.android.tools.build:gradle:2.3.0

In the meantime, our team will work on removing any need for jcenter dependency, but as of now this should enable you to continue working on this.

**Comment by ziga-hvalec** [comment]
I don't have this. How would I go around doing this in you package and then build on CI?

**Comment by raxit97** [comment]
Faced the same issue. 
As @KolissnikBogdan suggested, updated the gradle plugin version in the node_modules/react-native-ux-cam/android folder to 2.3.0 and added a patch to the repo using the following command:
`npx patch-package react-native-ux-cam`
Now its working.

**Comment by TrekSoft** [comment]
Yeah this is the error we're getting when we try to build on AppCenter - updating the version locally doesn't work because of course AppCenter is not pulling the locally changed code:

```
A problem occurred configuring project ':react-native-ux-cam'.
> Could not resolve all artifacts for configuration ':react-native-ux-cam:classpath'.
   > Could not find com.android.tools.build:gradle:2.3.1.
     Searched in the following locations:
       - https://jcenter.bintray.com/com/android/tools/build/gradle/2.3.1/gradle-2.3.1.pom
     If the artifact you are trying to retrieve can be found in the repository but without metadata in 'Maven POM' format, you need to adjust the 'metadataSources { ... }' of the repository declaration.
     Required by:
         project :react-native-ux-cam
```

EDIT: After removing UXCam as a dependency, a bunch of other libraries also showed up as failing so this seems to just be a pretty general outage.

**Comment by djob195** [comment]
It is not working...

`npx patch-package react-native-ux-cam
patch-package 6.5.0
• Creating temporary folder
• Installing react-native-ux-cam@5.4.7 with yarn
• Diffing your files with clean files
⁉️  Not creating patch file for package 'react-native-ux-cam'
⁉️  There don't appear to be any changes.`

* What went wrong:
A problem occurred configuring project ':react-native-ux-cam'.
> Could not determine the dependencies of null.
   > Could not resolve all task dependencies for configuration ':react-native-ux-cam:classpath'.
      > Could not find com.android.tools.build:gradle:2.3.1.

**Comment by jeanhipages** [comment]
> Faced the same issue. As @KolissnikBogdan suggested, updated the gradle plugin version in the node_modules/react-native-ux-cam/android folder to 2.3.0 and added a patch to the repo using the following command: `npx patch-package react-native-ux-cam` Now its working.

We have done the same and the android build is working in CI again.  @gabrieluxcam please ping this thread once the issue is resolved in uxcam so we can remove the patch 🙏

**Comment by rafaelgarcia13** [comment]
A solution that worked for me, was add mavenCentral on buildscript repositories

![image](https://github.com/uxcam/react-native-ux-cam/assets/64671122/947eb162-716d-4cb4-96b4-c9adc5211dcc)

**Comment by gabrieluxcam** [comment]
Hello @jeanhipages @djob195 @TrekSoft @raxit97 @ziga-hvalec @KolissnikBogdan We have now released React Native UXCam SDK version 5.4.8 which has updated the build gradle to avoid similar issues in the future.

dependency removed: jcenter
dependency added:  google(),  mavenCentral()
Updated build gradle to com.android.tools.build:gradle:7.2.2 from com.android.tools.build:gradle:2.3.1

Please do let us know if this continues to exist. I will mark this as closed for the time being.

**Comment by thisisadarshjais** [comment]
Facing the same issue for `react-native:"0.72.3"` with `"react-native-ux-cam": "5.4.8",

Could not determine the dependencies of task ':react-native-ux-cam:compileDebugJavaWithJavac'.
> Could not determine the dependencies of null.
   > Cannot query the value of this provider because it has no value available.
   
this is my gradel configuration

buildscript {
    ext {
            buildToolsVersion = "33.0.0"
            minSdkVersion = 21
	        compileSdkVersion = 33
	        targetSdkVersion = 33
	        // We use NDK 23 which has both M1 support and is the side-by-side NDK version from AGP.
            ndkVersion = "23.1.7779620"
            androidXCore = "1.7.0" // <-- Add this. Check versions here: https://developer.android.com/jetpack/androidx/releases/core
    }
    repositories {
        google()
        mavenCentral()
      
    }
    dependencies {
        classpath('com.android.tools.build:gradle')
        classpath("com.facebook.react:react-native-gradle-plugin:1.5.31")
        classpath("com.google.gms:google-services:4.3.10")  // Google Services plugin
       // classpath 'com.guardsquare:proguard-gradle:7.2.1'  // The ProGuard Gradle plugin.



        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
}


allprojects {
    repositories {
       maven { url 'https://maven.zohodl.com' }  
       google()
       mavenCentral()
    }
 }
